### PR TITLE
[app_dart] Add BuildStatus badge endpoint

### PR DESCRIPTION
**Pull request by CaseyHillers**
* Enable us to track Flutter CI in the READMEs

image

image

https://github.com/flutter/flutter/issues/140460

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

### CODE REVIEW

File: app_dart/bin/server.dart [hunk 1]
```diff
@@ -201,6 +201,12 @@ Future<void> main() async {
         delegate: GetBuildStatus(config: config),
         ttl: const Duration(seconds: 15),
       ),
+      '/api/public/build-status-badge': CacheRequestHandler<Body>(
```
Comment (keyonghan) [inline app_dart/bin/server.dart:204]: Could you add a CODEOWNER entry for this API?

**Review by keyonghan — APPROVED** [review]
LGTM with a nit.

File: CODEOWNERS [hunk 2]
```diff
@@ -11,6 +11,7 @@ app_dart/lib/src/request_handlers/dart_internal_subscription.dart           @dre
 app_dart/lib/src/request_handlers/file_flaky_issue_and_pr.dart              @keyonghan
 app_dart/lib/src/request_handlers/flush_cache.dart                          @keyonghan
 app_dart/lib/src/request_handlers/get_build_status.dart                     @keyonghan
+app_dart/lib/src/request_handlers/get_build_status.dart                     @CaseyHillers
```
Comment (keyonghan) [inline CODEOWNERS:14]: Seems a duplicated entry was added.

Comment (CaseyHillers) [reply CODEOWNERS:14]: Whoops, thanks for the catch! Sent https://github.com/flutter/cocoon/pull/3366

**Pull request merged**